pkg/service/service.go

```diff
@@ -953,20 +955,28 @@ func (s *Service) GetDeepCopyServicesByName(name, namespace string) (svcs []*lb.
 
 // RestoreServices restores services from BPF maps.
 //
+// It first restores all the service entries, followed by backend entries.
+// In the process, it deletes any duplicate backend entries that were leaked, and
+// are not referenced by any service entries.
+//
 // The method should be called once before establishing a connectivity
 // to kube-apiserver.
 func (s *Service) RestoreServices() error {
 	s.Lock()
 	defer s.Unlock()
+	var errs error
+	backendsById := make(map[lb.BackendID]struct{})
 
-	// Restore backend IDs
-	if err := s.restoreBackendsLocked(); err != nil {
-		return err
+	// Restore service cache from BPF maps
+	if err := s.restoreServicesLocked(backendsById); err != nil {
+		errs = multierr.Append(errs,
```

Comment:
nit: I don't think this ever actually returns a non-nil error value.

Reply:
sorry should've specified the actual line, restoreServicesLocked doesn't ever appear to return anything but nil for its error.